graphs/possible_bipartition.py

```diff
@@ -8,5 +8,63 @@ def possible_bipartition(dislikes):
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+
+def possible_bipartition(dislikes):
+    """ Will return True or False if the given graph
+        can be bipartitioned without neighboring nodes put
+        into the same partition.
+        Time Complexity: ?
+        Space Complexity: ?
+    """
+    if len(dislikes) == 0:
+        return True
+
+    groups = {key: 0 for key in dislikes.keys()}
+
+    first_dog = list(dislikes.keys())[0]
+    groups[first_dog] = 1
+    queue = deque([first_dog])
+    visited = set()
+
+    while queue:
+        current = queue.popleft()
+        visited.add(current)
+
+        if dislikes[current]:
+            for disliked_dog in dislikes[current]:
+                if groups[disliked_dog] == 0:
+                    groups[disliked_dog] = groups[current] + 1
+                    queue.append(disliked_dog)
+                else:
+                    if groups[disliked_dog] == groups[current]:
+                        return False
+
+    for dog in dislikes.keys():
+        if dog not in visited:
+            queue.append(dog)
+            groups[dog] = 1
+            while queue:
+                current = queue.popleft()
+                visited.add(current)
+                for disliked_dog in dislikes[current]:
+                    if disliked_dog not in visited:
```

Comment:
You were really close, just needed to bring the code from your first attempt on line 35 into this implementation

It's not enough to check that the dog isn't in visited before increasing the . The `disliked_dog` (neighbor of `current`) could have already been placed in group 1 because of another one of it's neighbors.
```suggestion
                    if disliked_dog not in visited and groups[disliked_dog] == 0:
```